packages/shape/src/dict.ts

```diff
@@ -10,22 +10,24 @@ import {
 
 export type DictShorthand = { [key: string]: Shorthand };
 
+type Internal<S extends DictShorthand> = {
+  Definition: { -readonly [K in keyof S]: DefinitionOf<S[K]> };
+  Serialized: {
+    -readonly [K in keyof S]: ReturnType<DefinitionOf<S[K]>["$serialize"]>;
+  };
+  Inline: {
+    -readonly [K in keyof S]: DefinitionOf<S[K]>["$inline"];
+  };
+};
+
 export const Dict = <
   const S extends { [key: string]: any },
   B extends AbstractConstructor<{}> = typeof Empty,
+  Cache extends Internal<S> = Internal<S>,
```

Comment:
cant you define Cache as a scoped type instead of this "hack"?

Reply:
Scoped variables result in inlined types when transpiled, this reduces the typescript serialization size by a lot, allowing me to fix some issues I had with big shapes